web/src/pages/Service/Create.tsx

```diff
@@ -0,0 +1,138 @@
+
+  const STEP_HEADER = [
+    formatMessage({ id: 'page.service.steps.stepTitle.basicInformation' }),
+    formatMessage({ id: 'page.service.steps.stepTitle.pluginConfig' }),
+    formatMessage({ id: 'component.global.steps.stepTitle.preview' }),
+  ]
+
+  const [stepHeader] = useState(STEP_HEADER);
+  const [step, setStep] = useState(1);
+
+  useEffect(() => {
+    const { serviceId } = (props as any).match.params;
+    if (serviceId) {
+      fetchItem(serviceId).then(({ data }) => {
+        if (data.upstream_id && data.upstream_id !== '') {
+          upstreamForm.setFieldsValue({ upstream_id: data.upstream_id });
+        }
+        if (data.upstream) {
+          upstreamForm.setFieldsValue(data.upstream);
+        }
+        form.setFieldsValue(omit(data, ['upstream_id', 'upstream', 'plugins']));
+        setPlugins(data.plugins || {});
+      });
+    }
+  }, []);
+
+  const onSubmit = () => {
+    const data = {
+      ...form.getFieldsValue(),
+      plugins,
+    };
+
+    const upstreamFormData = upstreamForm.getFieldsValue();
+    if (upstreamFormData.upstream_id === '') {
+      data.upstream = omit(upstreamFormData, ['upstream_id']);
+    } else {
+      data.upstream_id = upstreamFormData.upstream_id;
+    }
+
+    const { serviceId } = (props as any).match.params;
+    (serviceId ? update(serviceId, data) : create(data))
+      .then(() => {
+        notification.success({
+          message: `${serviceId
+            ? formatMessage({ id: 'component.global.edit' })
+            : formatMessage({ id: 'component.global.create' })
+            } ${formatMessage({ id: 'menu.service' })} ${formatMessage({
+              id: 'component.status.success',
+            })}`,
+        });
+        history.push('/service/list');
+      })
+      .catch(() => {
+        setStep(3);
+      });
+  };
+
+  const onStepChange = (nextStep: number) => {
+    if (step === 1 && nextStep === 2) {
+      form.validateFields().then(() => {
+        upstreamForm.validateFields().then(() => {
+          setStep(nextStep);
+        })
+      })
+      return;
+    }
+    if (nextStep === 4) {
+      onSubmit();
+      return;
+    };
+    setStep(nextStep);
+  }
+
+  return (<>
+    <PageHeaderWrapper
+      title={`${(props as any).match.params.rid
+        ? formatMessage({ id: 'component.global.edit' })
+        : formatMessage({ id: 'component.global.create' })
+        } ${formatMessage({ id: 'menu.service' })}`}
+    >
+      <Card bordered={false}>
+        <Steps current={step - 1} style={{ marginBottom: "25px" }}>
+          {stepHeader.map((item) => (
+            <Step title={item} key={item} />
+          ))}
+        </Steps>
+        {step === 1 && <Step1
+          form={form}
+          upstreamForm={upstreamForm}
+          upstreamRef={upstreamRef}
+        />}
+        {step === 2 && (
+          <PluginPage initialData={plugins} onChange={setPlugins} schemaType="route" />
```

Comment:
just make sure we do need to add `plugin pipeline` in service module，right？

Reply:
you mean `plugin orchestration` right?  We currently don't add plugin orchestration.
refer to https://github.com/apache/apisix/blob/560a2fe3ab49821f7eb90e21450839019893fe42/doc/zh-cn/admin-api.md#service

Reply:
got it